### PR TITLE
fix: [FIND-123] Zero-Amount Hold Execution Creates Partition That Blocks fullRedeemAtMaturity

### PR DESCRIPTION
**Pull request by jaime-iobermudez**
This pull request introduces stricter input validation across the asset contract domain to prevent zero-amount operations for holds, locks, and freezes. These changes ensure that attempts to create holds, locks, or freezes with a zero amount are rejected early, reducing the risk of edge-case bugs and denial-of-service vectors. The update also includes new custom error types, updated Solidity interfaces, and comprehensive registry and test coverage for these checks.

**Input validation and error handling improvements:**

* Added custom errors `InvalidHoldAmount`, `InvalidLockAmount`, and `InvalidFreezeAmount` to the respective interfaces (`IHoldTypes`, `ILockTypes`, `IFreeze`) to standardize error handling for zero-amount operations. [[1]](diffhunk://#diff-0741681e71f0ed0107af103ac3e80ae3a567028a531465dd914d3aff3387286eR116) [[2]](diffhunk://#diff-addbb37935eda517cf0c6b44524b85c12c973cdb25628e59e0e0381dae8f718aR69-R75) [[3]](diffhunk://#diff-dc206dad649cb459f8a47335d1ea35d7b3807a18501d3fd6be5b34231c4eb555R42-R49)
* Enforced non-zero amount checks at the start of hold, lock, and freeze creation functions in `HoldStorageWrapper`, `LockStorageWrapper`, and `ERC3643StorageWrapper`, reverting with the new errors if violated. [[1]](diffhunk://#diff-9d0d744e7cebc6a0e0957d1d99644b486542b317400fb6cd0b3014e811e10ef5R55-R56) [[2]](diffhunk://#diff-9d0d744e7cebc6a0e0957d1d99644b486542b317400fb6cd0b3014e811e10ef5R515-R523) [[3]](diffhunk://#diff-5afa74df0a0ad47409d4e2fc5f06826ec66a5770efb812bc17807b843912e40cR231-R232) [[4]](diffhunk://#diff-2ea5fc7e64996ffe53d049f2dbe1e3274df7ff48a1743bbb186d0514d990ea12R43-R44) [[5]](diffhunk://#diff-7a741b116a06ea131933f2a1986e2471cdee7282ac578e5967cc092db9d14cf3R139-R140)
* Updated the `ClearingOps` library to reuse the centralized zero-amount hold check.

**Registry and interface updates:**

* Registered the new custom errors in the facet registry data (`atsRegistry.data.ts`), ensuring correct ABI and selector mappings for `InvalidHoldAmount`, `InvalidLockAmount`, and `InvalidFreezeAmount` across all relevant facets. [[1]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R968-R972) [[2]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R1492-R1496) [[3]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R3843-R3847) [[4]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R6777-R6781) [[5]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R7157-R7161) [[6]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R7357-R7361) [[7]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R7601-R7605) [[8]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R8671-R8675) [[9]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R8893-R8897) [[10]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R10477-R10481) [[11]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R11787-R11791) [[12]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R12158-R12162) [[13]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R13108-R13112) [[14]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R13249-R13253) [[15]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R13389-R13393) [[16]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R13529-R13533) [[17]](diffhunk://#diff-4877815dbe43a7ccce353957f47744602330bcb60863d6b811363d6e886cd949R13669-R13673)

**Testing and documentation:**

* Added an integration test to verify that attempting to create a hold with zero amount reverts with `InvalidHoldAmount`, preventing ghost partition DoS attacks in `fullRedeemAtMaturity`.
* Updated interface and function documentation to clarify the semantics and rationale for rejecting zero-amount operations. [[1]](diffhunk://#diff-9d0d744e7cebc6a0e0957d1d99644b486542b317400fb6cd0b3014e811e10ef5R515-R523) [[2]](diffhunk://#diff-dc206dad649cb459f8a47335d1ea35d7b3807a18501d3fd6be5b34231c4eb555R42-R49) [[3]](diffhunk://#diff-addbb37935eda517cf0c6b44524b85c12c973cdb25628e59e0e0381dae8f718aR69-R75)

**Minor code cleanup:**

* Removed unused imports and simplified logic in the `Maturity` contract, now only redeeming partitions with non-zero balances. [[1]](diffhunk://#diff-c2454de09e3856738b6f56be23519a9bd25f8267813cfd47cf6ecfb8212a44e3L7-L13) [[2]](diffhunk://#diff-c2454de09e3856738b6f56be23519a9bd25f8267813cfd47cf6ecfb8212a44e3L48-R48)
* Updated facet registry generation timestamp to reflect the latest contract changes.

## Type of change

- [ ] Bug fix 🐞
- [ ] New feature ✨
- [ ] Breaking change 💥
- [ ] Documentation update 📖
- [ ] Refactor 🔧

## Testing

**Node version**:

- [ ] 20
- [ ] 22
- [ ] 24

## Checklist

- [ ] Style Guidelines followed ✅
- [ ] Documentation Updated 📚
- [ ] **Linters** - No New Warnings ⚠️
- [ ] Local Tests Pass ✅
- [ ] Effective Tests Added ✔️
- [ ] No reduction of **Coverage** ✅

### CODE REVIEW

File: packages/ats/contracts/contracts/domain/asset/LockStorageWrapper.sol [hunk 1]
```diff
@@ -40,6 +40,8 @@ library LockStorageWrapper {
         uint256 expirationTimestamp,
         address operator
     ) internal returns (bool success_, uint256 lockId_) {
+        if (amount == 0) revert ILockTypes.InvalidLockAmount();
```
Comment (marcosio) [inline packages/ats/contracts/contracts/domain/asset/LockStorageWrapper.sol]: Use check function.

File: packages/ats/contracts/contracts/domain/core/ERC3643StorageWrapper.sol [hunk 2]
```diff
@@ -135,6 +136,8 @@ library ERC3643StorageWrapper {
     }
 
     function freezeTokensByPartition(bytes32 _partition, address _account, uint256 _amount) internal {
+        if (_amount == 0) revert IFreeze.InvalidFreezeAmount();
```
Comment (marcosio) [inline packages/ats/contracts/contracts/domain/core/ERC3643StorageWrapper.sol]: Use check function.

**Review by marcosio — CHANGES_REQUESTED** [review]

**Review by marcosio — APPROVED** [review]
